packages/core/src/index.ts

```diff
@@ -96,6 +96,34 @@ function batch<T>(callback: () => T): T {
 	}
 }
 
+// Each nested transaction stores snapshots of the signals it changed.
+const transactionSnapshots: Array<Map<Signal<unknown>, [number, unknown]>> = [];
```

Comment:
I used a map for simplicity but it seems this library avoids them. I can refactor it to a plain object if there's a preference.